src/Admin/FieldDescription.php

```diff
@@ -93,19 +94,53 @@ public function isIdentifier()
     public function getValue($object)
     {
         foreach ($this->parentAssociationMappings as $parentAssociationMapping) {
-            $object = $this->getFieldValue($object, $parentAssociationMapping['fieldName']);
+            $object = $this->getChildValue($object, $parentAssociationMapping['fieldName']);
         }
 
+        /*
+         * NEXT_MAJOR: Remove this code and the related tests.
+         * The support for embedded object come from the next foreach if the fieldName is correctly set.
+         */
         $fieldMapping = $this->getFieldMapping();
         // Support embedded object for mapping
         // Ref: https://www.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/embeddables.html
         if (isset($fieldMapping['declaredField'])) {
             $parentFields = explode('.', $fieldMapping['declaredField']);
             foreach ($parentFields as $parentField) {
-                $object = $this->getFieldValue($object, $parentField);
+                $object = $this->getChildValue($object, $parentField);
             }
         }
 
-        return $this->getFieldValue($object, $this->fieldName);
+        $fields = explode('.', $this->getFieldName());
+        $lastField = array_pop($fields);
+        foreach ($fields as $field) {
+            $object = $this->getChildValue($object, $field);
+        }
+
+        return $this->getFieldValue($object, $lastField);
+    }
+
+    /**
+     * @throws NoValueException
+     */
+    private function getChildValue(?object $object, string $fieldName): ?object
+    {
+        if (null === $object) {
+            return null;
+        }
+
+        $child = $this->getFieldValue($object, $fieldName);
+        if (null !== $child && !\is_object($child)) {
+            throw new NoValueException(sprintf(
+                'Unexpected value when accessing to the property "%s" on the class "%s" for the field "%s".'
+                .' Expected object|null, got %s.',
```

Comment:
If you need to split the string in the code, it's probably too long in the output exception message. How about using linkebreaks in the code AND in the output?

```php
<<<'EXCEPTION'
Unexpected value when accessing to the property "%s" on the class "%s" for the field "%s".
Expected object|null, got %s.
EXCEPTION
```